### PR TITLE
Remove migrated JAXRS tests

### PR DESCRIPTION
**Pull request by alwin-joseph**
**Fixes Issue**
No issue created

**Describe the change**
The standalone JAXRS TCK tests, documentation and related contents which were migrated to https://github.com/jakartaee/rest are removed from jakartaee-tck. 
- Also updated the jakartaee-tck/docker/jaxrstck/pom.xml file that is used to run new Jakarta REST TCK against Glassfish.
- Retained below platform level tests (and dependent files) to be run in Platform TCK as before.
src/com/sun/ts/tests/jaxrs/jaxrs21/platform
src/com/sun/ts/tests/jaxrs/platform

In a separate change Platform TCK requirements will be modified to also run this new Jakarta REST TCK for platform compatibility.

### CODE REVIEW

**Comment by scottmarlow** [comment]
Thanks @alwin-joseph for the clear description of what you changed.  I will look at the changes locally.

File: src/com/sun/ts/tests/jaxrs/common/provider/StringBeanRuntimeDelegate.java
Comment (scottmarlow) [inline src/com/sun/ts/tests/jaxrs/common/provider/StringBeanRuntimeDelegate.java]: What is this test verifying?  In the new REST TCK, I see this class being referenced from other tests but I don't see any references to StringBeanRuntimeDelegate in this pull request.

Comment (alwin-joseph) [reply src/com/sun/ts/tests/jaxrs/common/provider/StringBeanRuntimeDelegate.java]: You are right, it is not used. There are more files under src/com/sun/ts/tests/jaxrs/common that can be removed. I will make that change.

File: docker/jaxrstck/README
Comment (scottmarlow) [inline docker/jaxrstck/README:6]: FYI for others reading this pull request, the files in [docker/jaxrstck](https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/docker/jaxrstck) are not distributed (currently) with any TCKs, instead these notes and maven pom is for verifying that the REST TCK passes with GlassFish.  IMO, `pom.xml` for other compatible implementations besides GlassFish could be added in [this folder](https://github.com/eclipse-ee4j/jakartaee-tck/tree/master/docker/jaxrstck) as well (if other RESTful Web Service implementations choose to create a pull request for doing so before EE 10 goes final).  

CC @brideck @jamezp ^

**Comment by jansupol** [comment]
Leaving the following folders looks good for platform TCK:
- src/com/sun/ts/tests/jaxrs/jaxrs21/platform
- src/com/sun/ts/tests/jaxrs/platform

**Comment by alwin-joseph** [comment]
> As mentioned above, email should be sent to the [Platform mailing list](jakartaee-platform-dev@eclipse.org), [Platform TCK mailing list ](jakartaee-tck-dev@eclipse.org) and [SPEC API](jaxrs-dev@eclipse.org) mailing list. Anyone can send that, let me know if you would like me to send it. We should request that feedback/approval be left on this issue tracker instead of discussing on the email list.

@scottmarlow Thanks! Can you please send that email asking for feedback from Platform and spec api teams.

**Comment by spericas** [comment]
I will defer to @alwin-joseph and @jansupol to review the details of this one. Removing these tests seems like the necessary to complete the migration.

**Comment by scottmarlow** [comment]
I suggest that we merge this change tomorrow if there are no objections by then.

**Review by gurunrao — APPROVED** [review]

**Comment by alwin-joseph** [comment]
@jansupol @scottmarlow @gurunrao . I have verified and checked the below.

> 
>     * [x]  Platform TCK pull request is needed to eliminate any Platform TCK tests not needed to validate Platform SPEC requirements.
> 
>     * [x]  Send email to [Platform](https://accounts.eclipse.org/mailing-list/jakartaee-platform-dev), [Platform TCK](https://accounts.eclipse.org/mailing-list/jakartaee-tck-dev), relevant SPEC API mailing lists with PR link for removing duplicate tests.
> 
>     * [x]  Ensure that some validation of SPEC API is done for each tested EE container/vehicle.
> 
>     * [x]  Review each test and make a determination if there is a specific Platform requirement being tested beyond just using any parts of SPEC API in specified containers.
> 
>     * [x]  If the new SPEC API TCK validates the Platform requirements (with required EE container/vehicles), all of the Platform TCK tests specific to that SPEC API may be removed.
> 
> 

> Leaving the following folders looks good for platform TCK:
> 
>     * src/com/sun/ts/tests/jaxrs/jaxrs21/platform
> 
>     * src/com/sun/ts/tests/jaxrs/platform

Thanks @jansupol ! I believe there were no new jaxrs tests for Platform that needed to be created for Jaxrs 3.1 , if so they would have been added here in these folders.

We can merge this if no other comments and feedback.

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**
